HELM 386:Chart Verifier api integration (#11964)
openshift/console: docs/helm/helm-chart-verification-via-console.md

# Helm Chart Verification

> Chart Verifier is a cli tool that runs a series of independent checks defined in different profiles. This CLI tool is used on our Chart Certification Flow.

Reviewer: ```suggestion
Chart-verifier is a CLI tool that runs a series of independent checks defined in different profiles. This CLI tool is used to validate the Helm chart against a configurable list of checks through an automated Red Hat OpenShift certification workflow. Certification assures the integrity of the chart and ensures that the Helm chart works seamlessly on Red Hat OpenShift clusters.
```

> The Helm Chart Verifier api `/api/helm/verify` is an integration of this cli tool  on `Openshift Developer Console`(ODC) and this will provide user more insight on whether the helm charts they are targeting will run well on OpenShift. The following checks will be run when user is looking to verify their helm chart from console:

Reviewer: ```suggestion
The `/api/helm/verify` Helm chart-verifier API integrates the chart-verifier CLI tool on Openshift Developer Console (ODC). This API provides users more insight into whether the Helm charts that they are targeting will run well on OpenShift. The following checks are run when the user needs to verify their Helm chart from the console:
```

1. ContainsTest: Checks chart test files exist
2. ContainsValuesSchema: Checks whether values schema file exists
3. ContainsValues: Checks if values file exist
4. HasKubeVersion: Checks kubernetes version specified
5. HasReadme: Checks chart has a README
6. HelmLint: checks Helm lint for the chart
7. ImagesAreCertified: Checks if image is Red Hat certified 
8. IsHelmV3: Checks the apiversion for the chart and whether it is used in Helm 3
9. NotContainCsiObjects: Checks whether CSI objects are present
10. NotContainsCRDs: Checks chart does not contain CRDs
11. RequiredAnnotationsPresent: Checks all required annotations are present

> 

Reviewer: Suggestion: Put in table format with headers as Checks and Description (2 columns)

@Kartikey-star We Will sit together to review this tomorrow when you are available.

Reviewer: @Kartikey-star Please follow the descriptions written here: https://github.com/openshift/console/pull/11964/files#r962695319

> As you can see a chart which goes through the certification process verifies that the  Helm chart is compliant with these set of independent checks and is recommended to be used.

Reviewer: ```suggestion
A chart that goes through the certification process verifies that the Helm chart is compliant with the predefined set of independent checks and is recommended for use.
```

> The output of the chart verifier api is JSON encoded [Release structure](https://github.com/redhat-certification/chart-verifier/blob/main/pkg/chartverifier/report/types.go#L12-18) 

Reviewer: ```suggestion
The output of the char-verifier API is JSON encoded [Release structure](https://github.com/redhat-certification/chart-verifier/blob/main/pkg/chartverifier/report/types.go#L12-18).
```
Suggestion: Probably needs a rewrite for more clarity.

## Additional Resources

> To read more about `Chart Verifier` please go through  [RedHat Chart Verifier](https://github.com/redhat-certification/chart-verifier).

Reviewer: ```suggestion
For more information on `chart-verifier`, see [RedHat Chart Verifier](https://github.com/redhat-certification/chart-verifier).
```

| IsHelmV3 | Checks that the given `uri` points to a Helm v3 chart. |
| HasReadme | Checks that the Helm chart contains the `README.md` file. |
| ContainsTest | Checks that the Helm chart contains at least one test file. |
| HasKubeVersion | Checks that the `Chart.yaml` file of the Helm chart includes the `kubeVersion` field (v1.0) and is a valid semantic version (v1.1). |
| ContainsValuesSchema | Checks that the Helm chart contains a JSON schema file (`values.schema.json`) to validate the `values.yaml` file in the chart. |
| NotContainsCRDs | Checks that the Helm chart does not include custom resource definitions (CRDs). |
| NotContainCsiObjects |  Checks that the Helm chart does not include Container Storage Interface (CSI) objects. |
| ImagesAreCertified | Checks that the images referenced by the Helm chart are Red Hat-certified .|
|  HelmLint | Checks that the chart is well formed by running the `helm lint` command. |
| ContainsValues | Checks that the Helm chart contains the `values`[¹](https://github.com/redhat-certification/chart-verifier/blob/main/docs/helm-chart-checks.md#-for-more-information-on-the-values-file-see-values-and-best-practices-for-using-values) file. |

> | RequiredAnnotationsPresent |Checks that the Helm chart contains the annotation: ```charts.openshift.io/name```. |

Reviewer 2: Is there a chart already in chart verifier docs we can reference?

Author: You mean to refer a chart which passes all the checks?

A chart that goes through the certification process verifies that the Helm chart is compliant with the predefined set of independent checks and is recommended for use.

> The output of the chart-verifier API is a JSON-encoded [response ](https://github.com/redhat-certification/chart-verifier/blob/main/pkg/chartverifier/report/types.go#L12-18).

Reviewer 2: I think we should talk about profiles here, I would create a profile for console under chart verifier. We default to that if api client does not specify the profile.
